### DISCUSSION TITLE
RE: fatal: The current branch master has no upstream branch. | and also 'https://gitbuh.com/MyName'

### DISCUSSION THREAD

**Discussion by AnneMarieHarm** [community/community#23239] [New to GitHub]
<p>Hello, I joined Github last week, created and initiated my first two repositories. Now I want to add, commit and push revised files to those repositories. When I pushed the revised files, I received the following response: ----------------------------------------- $ git push fatal: The current branch master has no upstream branch. To push the current branch and set the remote as upstream, use git push --set-upstream origin master ------------------------------------------ So I did, and received response: My Name@DESKTOP-32QET7M TINGS64 ~/Desktop (master) — (I revised this line, with same type of characters) $ git push --set-upstream origin master fatal: unable to access ‘<a href="https://gitbuh.com/AnneMarieHarm/deliverable_one.git/" rel="nofollow noopener">https://gitbuh.com/AnneMarieHarm/deliverable_one.git/</a>’: Could not resolve host: <a href="http://gitbuh.com" rel="nofollow noopener">gitbuh.com</a> ------------------------------------ I wonder how it became ‘<a href="http://gitbuh.com" rel="nofollow noopener">gitbuh.com</a>’ (not ‘<a href="http://github.com" rel="nofollow noopener">github.com</a>’)? And why I can’t seem to push revised files to my respositories? Any help anyone may be able to provide, would be appreciated. I’ve reviewed help files and references, and found this other user receiving some of the same responses (<a href="https://github.community/t/pushing-to-collaborators-private-repo-from-terminal/654">https://github.community/t5/How-to-use-Git-and-GitHub/Pushing-to-collaborator-s-private-repo-from-terminal/m-p/4170</a> VishAlSahim Ground Controller Lvl 1 ‎02-23-2018 07:55 PM Message 1 of 2 Pushing to collaborator’s private repo from terminal: ) ------------------------------------------- However VishAlSahim’s situation seems more complex, as he is trying to collaborate by using or contributing to someone else’s repository. Thank you!</p>

**Comment by strafe** [comment, marked as answer]
<p>Hi <a class="mention" href="/annemarieharm">@annemarieharm</a>, you can ammend your remote URL with the following command:</p>
<pre><code>git remote set-url origin https://github.com/AnneMarieHarm/deliverable_one.git
</code></pre>

**Reply by sajjantimalsina123** [reply]
git remote set-url origin https://github.com/AnneMarieHarm/deliverable_one.git

**Comment by AnneMarieHarm** [comment]
<p>Hi strafe, </p>
<p>I appreciate the reply; that is the same thing Stacey from Github Support recommended I do, with the addition of ‘.git’ at the end (in your case). I also see this referenced in the article she referred me to (<a href="https://help.github.com/articles/changing-a-remote-s-url/" rel="nofollow noopener">https://help.github.com/articles/changing-a-remote-s-url/</a>) – complete with .git at the end. However, this has not worked in my case; I’ve pasted recent results of that effort, below my signature. Any other options come to mind?</p>
<h2>Thank you. ~ Anne Marie</h2>
<p>Anne Marie Harm@DESKTOP-32QE01T MINGW64 ~/Desktop (master)<br>
$ git remote set-url origin <a href="https://github.com/AnneMarieHarm/deliverable_one.git" rel="nofollow noopener">https://github.com/AnneMarieHarm/deliverable_one.git</a></p>
<p>Anne Marie Harm@DESKTOP-32QE01T MINGW64 ~/Desktop (master)<br>
$ git push<br>
fatal: The current branch master has no upstream branch.<br>
To push the current branch and set the remote as upstream, use</p>
<p>git push --set-upstream origin master</p>
<p>Anne Marie Harm@DESKTOP-32QE01T MINGW64 ~/Desktop (master)<br>
$ git push --set-upstream origin master<br>
git remote set-To <a href="https://github.com/AnneMarieHarm/deliverable_one.git" rel="nofollow noopener">https://github.com/AnneMarieHarm/deliverable_one.git</a><br>
! [rejected] master -&gt; master (fetch first)<br>
error: failed to push some refs to ‘<a href="https://github.com/AnneMarieHarm/deliverable" rel="nofollow noopener">https://github.com/AnneMarieHarm/deliverable</a> _one.git’<br>
hint: Updates were rejected because the remote contains work that you do<br>
hint: not have locally. This is usually caused by another repository pushing<br>
hint: to the same ref. You may want to first integrate the remote changes<br>
hint: (e.g., ‘git pull …’) before pushing again.<br>
hint: See the ‘Note about fast-forwards’ in ‘git push --help’ for details.</p>

**Comment by strafe** [comment]
<p>Hi <a class="mention" href="/annemarieharm">@annemarieharm</a>, I’m glad your first issue is resolved. You can see the hints tell you there’s commits on the remote that you don’t have locally, this means it’s preventing you form pushing so you don’t erase those commits from the remote. Assuming there aren’t going to be any conflicts you should be able to use:</p>
<pre><code>git pull origin master
</code></pre>
<p>followed by:</p>
<pre><code>git push
</code></pre>

**Comment by AnneMarieHarm** [comment]
<p>Hi strafe,<br>
Again I appreciate your response. I tried that, and received the following:</p>
<p>Anne Marie Harm@DESKTOP-32QE01T MINGW64 ~/Desktop (master)<br>
$ git pull origin master<br>
From <a href="https://github.com/AnneMarieHarm/deliverable_one" rel="nofollow noopener">https://github.com/AnneMarieHarm/deliverable_one</a><br>
* branch master -&gt; FETCH_HEAD<br>
fatal: refusing to merge unrelated histories</p>
<p>Anne Marie Harm@DESKTOP-32QE01T MINGW64 ~/Desktop (master)<br>
$ git push<br>
fatal: The current branch master has no upstream branch.<br>
To push the current branch and set the remote as upstream, use</p>
<h2>git push --set-upstream origin master</h2>
<p>I think next I’ll read up on doing a git pull, in case I haven’t done that correctly – and see what I find out. I do see that I may not have been in a good position to proceed to git push, from this message:</p>
<p>From <a href="https://github.com/AnneMarieHarm/deliverable_one" rel="nofollow noopener">https://github.com/AnneMarieHarm/deliverable_one</a><br>
* branch master -&gt; FETCH_HEAD<br>
fatal: refusing to merge unrelated histories</p>
<p>Thank you for your help. ~ Anne Marie</p>

**Comment by AnneMarieHarm** [comment]
<p>Hi Strafe,</p>
<p>I am grateful for your responses; it now looks like the git remote set-url origin <a href="https://github.com/AnneMarieHarm/deliverable_one" rel="nofollow noopener">https://github.com/AnneMarieHarm/deliverable_one</a> (with or without the .git file extension) resolved one of my issues (‘<a href="http://gitbuh.com" rel="nofollow noopener">gitbuh.com</a>’ typo). But it turns out while_ git pull_ may well also have allowed me to revise the files in question, I suspect it may have created others (and there was a simpler solution). Please let me know if you (or anyone else) would like further information.</p>
<p>Either way I hope to remember these lessons, and all I learned by A) hearing back from you (and Stacey from Github Support), and B) reviewing and studying related documentation – how the issues I experienced compelled me to become more familiar. And how nice it was, when it finally worked!</p>
<p>Thank you. ~ Anne Marie</p>

**Comment by legm7** [comment]
<p>Hi Anne Marie, I have the same problem and could not solve it!!</p>
<p>Thanks for your answer!!</p>

**Comment by Husnanaz** [comment]
<p>Hai Anne Marie. Currently I am facing the same problem with what you had. I could not push or pull and had the same messages. I was wondering if you could share the full solution? </p>
<p>Thank you. </p>

**Comment by Jean-Laurore** [comment]
<p>The problem is that you have to update your credential. When done updating your credential, use git push --set-upstream origin master for the first time after that you can use git push</p>
<p><a href="https://help.github.com/en/github/using-git/updating-credentials-from-the-osx-keychain" rel="nofollow noopener">https://help.github.com/en/github/using-git/updating-credentials-from-the-osx-keychain</a></p>

**Comment by AnneMarieHarm** [comment]
<p>I appreciate the response, and love the fact that I’m receiving it almost a year and a half after posting my question. It testifies to a great reality of online user forums, and the way shared knowledge and experience accrue. Thank you!</p>

**Comment by benreaves** [comment]
<p>You’re gonna hate me, but I solved it in a very ugly way. First, my context:</p>
<ol>
<li>I just finished moving a local repository to github by following <a href="https://help.github.com/en/github/importing-your-projects-to-github/adding-an-existing-project-to-github-using-the-command-line" rel="nofollow noopener">this</a>
</li>
<li>To check whether this did the trick, I open my browser to my github repository and locally make a small edit to README.md</li>
<li>I type<br>
git add README.md<br>
git commit -m . <br>
git push</li>
<li>I get the error<br>
fatal: The current branch master has no upstream branch.</li>
</ol>
<p><strong>Solution:</strong> I edited .git/config and added the last three lines:</p>
<pre><code>[core]
	repositoryformatversion = 0
	filemode = false
	bare = false
	logallrefupdates = true
	symlinks = false
	ignorecase = true
[remote "origin"]
	url = https://github.com/benreaves/whatever.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
</code></pre>
<p>It’s working for me. If any of you want to yell at me, please go ahead, but from a safe social distance of 2 meters :slight_smile:</p>

**Comment by AnneMarieHarm** [comment]
<p>I realize it’s a figure of speech, but I’ll never hate you. Glad you’re out there, doing what you do! Thank you.</p>

**Comment by ariessulistio** [comment]
<p><a href="https://blog.google/technology/safety-security/expediting-changes-google-plus/" rel="nofollow noopener">https://blog.google/technology/safety-security/expediting-changes-google-plus/</a></p>

**Comment by RllXeron** [comment]
<p>Thx it seems it helped!</p>

**Comment by mgates3** [comment]
<p>Had the same problem.<br>
<code>git branch --set-upstream-to origin/master</code> or<br>
<code>git branch -u origin/master</code><br>
adds those 3 lines to .git/config, per<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/22682994/181075710-3fe8196f-7504-4589-a503-c30d21525c0b.png" class="site-icon" width="200" height="200">
      <a href="https://devconnected.com/how-to-set-upstream-branch-on-git/" target="_blank" rel="nofollow noopener" title="05:55PM - 31 October 2019">devconnected – 31 Oct 19</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/388;"><img src="https://user-images.githubusercontent.com/22682994/181075711-34d4f266-94fe-463c-80e6-af3f61efbb52.jpeg" class="thumbnail" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/4f149c9a75860f4f30783bfc9a794a6806b8f2e6_2_690x388.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/4/4f149c9a75860f4f30783bfc9a794a6806b8f2e6_2_1035x582.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/4/4f149c9a75860f4f30783bfc9a794a6806b8f2e6_2_1380x776.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4f149c9a75860f4f30783bfc9a794a6806b8f2e6_2_10x10.png"></div>

<h3><a href="https://devconnected.com/how-to-set-upstream-branch-on-git/" target="_blank" rel="nofollow noopener">How To Set Upstream Branch on Git – devconnected</a></h3>

<p>Complete step-by-step tutorial on how to set upstream branches (remote tracking branches) on Git easily using branch commands.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by Sandeepad3118** [comment]
<p>Try this command works really well:<br>
git push origin master -f</p>

**Comment by madeelahsan** [comment]
<p>yup it’s works for me too.</p>

**Comment by francopagnotta** [comment]
<p>Hello! that works for me! I didn’t know how to solve the problem, but git push origin master -f was the solution! Can you tell me why? I would like to know ! I am new to Git and GitHub.</p>
